Sources/Redis/Client/RedisClient.swift

```diff
@@ -28,6 +31,12 @@ public final class RedisClient: DatabaseConnection, BasicWorker {
         self.extend = [:]
         self.isClosed = false
         channel.closeFuture.always {
+            // send closed error for the promises that have not been fulfilled
+            for promise in self.inflightPromises.values {
+                promise.fail(error: ChannelError.ioOnClosedChannel)
+            }
```

Comment:
This code could almost verbatim be moved to the new RequestResponseHandler, and attached during the `handlerAdded(ctx:)` protocol method without adding more memory usage.